### PR TITLE
Remove usage of BlackoilState class

### PR DESCRIPTION
**Pull request by totto82**
This allows (re)moving of the following files

opm/autodiff/RateConverterLegacy.hpp
opm/autodiff/Compat.cpp
opm/autodiff/Compat.hpp
opm/core/props/BlackoilPropertiesInterface.hpp
opm/core/simulator/BlackoilState.cpp
opm/core/simulator/BlackoilState.hpp
opm/core/simulator/BlackoilStateToFluidState.hpp
opm/core/utility/initHydroCarbonState.hpp
opm/polymer/PolymerBlackoilState.cpp
opm/polymer/PolymerBlackoilState.hpp
tests/test_blackoilstate.cpp

### CODE REVIEW

File: opm/autodiff/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -1545,4 +1545,70 @@ namespace Opm {
         }
     }
 
+
+    // convert well data from opm-common to well state from opm-core
+    template<typename TypeTag>
+    void
+    BlackoilWellModel<TypeTag>::
+    wellsToState( const data::Wells& wells,
+                       PhaseUsage phases,
```
Comment (totto82) [inline opm/autodiff/BlackoilWellModel_impl.hpp]: This is just copied from Compat.cpp

**Comment by totto82** [comment]
This PR needs a follow-up PR that moves the files in the list above to opm-legacy.  Can I trust you, 
@akva2, with this task?

**Comment by atgeirr** [comment]
I believe you are duplicating much of RateConverter.hpp into RateConverterEbos.hpp. Instead, I think you should extract the methods you want to get rid of to RateConverterLegacy.hpp, which would then be moved (it is better to keep the shorter name for the facility that stays in opm-simulators). It is then ok to let RateConverterLegacy depend on RateConverter for the utilities that are identical, rather than duplicating.

If I misunderstood anything, please correct me!

**Comment by akva2** [comment]
yes i will do the follow up, no worries.

**Comment by totto82** [comment]
To make legacy compile I added the no longer needed filenames in the CMakeLists_files.cmake. And now RateConverterLegacy must be moved instead of RateConverter as @atgeirr suggested. Needs a followup merge in OPM/opm-simulators-legacy#1

**Comment by totto82** [comment]
jenkins build this opm-simulators-legacy=1 please

**Comment by akva2** [comment]
there is no building of legacy as a downstream. just ignore it for now, i'll include your stuff when i do the moving.

**Comment by totto82** [comment]
@akva2 Ok, I have locally tested that legacy also build with PR #1. I will leave the rest to you. Please merge or hi-jack these commits or do whatever you like to finish this cleanup. Thanks!

**Comment by akva2** [comment]
yup. you should remove the installation of the legacy file from this pr, then merge this. then i'll do the rest.

**Comment by akva2** [comment]
continuied in #1661

**Pull request closed without merging**